components/playlist/browser/playlist_background_webcontents_helper.cc

```diff
@@ -0,0 +1,56 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/playlist/browser/playlist_background_webcontents_helper.h"
+
+#include "base/logging.h"
+#include "brave/components/playlist/common/mojom/playlist.mojom.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/web_contents.h"
+#include "third_party/blink/public/common/associated_interfaces/associated_interface_provider.h"
+#include "url/url_constants.h"
+
+namespace playlist {
+
+PlaylistBackgroundWebContentsHelper::~PlaylistBackgroundWebContentsHelper() =
+    default;
+
+void PlaylistBackgroundWebContentsHelper::ReadyToCommitNavigation(
+    content::NavigationHandle* navigation_handle) {
+  DVLOG(2) << __FUNCTION__;
+
+  DCHECK(navigation_handle);
+  if (!navigation_handle->IsInPrimaryMainFrame()) {
+    return;
+  }
+
+  const GURL url = navigation_handle->GetWebContents()->GetVisibleURL();
```

Comment:
The latest version is using `GetLastCommittedURL()` instead of `GetVisibleURL()` following https://github.com/brave/brave-core/pull/21332#discussion_r1487668333

Reply:
`GetLastCommittedURL()` is not the URL we want  in `ReadyToCommitNavigation()` — this navigation hasn't committed yet.

Reply:
Ah ha, I see, it's still a pending entry.

Reply:
Those are both incorrect. You should be using `navigation_handle->GetURL()` to get the url that is being navigated to. If there is any other code outside of this using `GetLastCommittedURL` or `GetVisibleURL` instead of the navigation handle it should also be fixed in a separate PR. 
https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/navigation_handle.h;l=100?q=navigation_handle.h&ss=chromium

Reply:
Fair enough — fixed.